shadow.c

```diff
@@ -555,7 +555,11 @@ static char *get_full_path(const char *filename TSRMLS_DC)
     	if(new_state.cwd) free(new_state.cwd);
         return NULL;
     }
-	return estrndup(new_state.cwd, new_state.cwd_length);
+	char *full_path = estrndup(new_state.cwd, new_state.cwd_length);
+	if (new_state.cwd) {
+		efree(new_state.cwd);
```

Comment:
I've changed `strdup` to `estrdup` instead. php-src encourages to use `e***` functions over standard, as they are used by engine for proper memory management (e.g. memory limit calculation).